runtime/commonMain/src/kotlinx/serialization/json/internal/JsonReader.kt

```diff
@@ -175,32 +175,58 @@ internal class JsonReader(private val source: String) {
         length += addLen
     }
 
+    fun nextJsonKey() {
```

Comment:
Can't you reuse `nextToken` with additional check `tokenClass == TC_OTHER || tokenClass == TC_STRING`?

Reply:
I am convinced that it is much more maintainable and readable to have two similar (but not exactly, not only in the additional check) methods

Reply:
What is an option is to have an inline function that implements the common code and then use parameters to split out the non-shared code. Something like:

```
inline fun nextTokenImpl(onElse: (Char) -> Unit, onEnd: () -> Unit) {
    val source = source
    val currentPosition = currentPosition
    val maxLen = source.length
    while (currentPosition<maxLen) {
        val ch = source[currentPosition]
        when(charToTokenClass(ch) {
            TC_WS -> CurrentPosition++ // skip whitespace
            TC_OTHER -> {
                nextLiteral(source, currentPosition)
                return
            }
            TC_STRING -> {
                nextLiteral(source, currentPosition)
                return
            }
            else -> {
                onElse(ch)
            }
        }
    }
    onEnd()
}

fun nextJsonKey() = nextTokenImpl(onElse = {
    fail("Expected next key name, but had trailing comma instead", currentPosition)
}, onEnd = {
    fail("Unexpected EOF", currentPosition)
})

fun nextToken() = nextTokenImpl(onElse = {
    this.tokenPosition = currentPosition
    this.tokenClass = tc
    this.currentPosition = currentPosition + 1
    return
}, onEnd = {
    tokenPosition = currentPosition
    tokenClass = TC_EOF
})
````

Reply:
Yes, it is possible. But I don't think that it makes the code more readable or understandable, rather the opposite. The ultimate goal here is not to write as much generic as possible code, but rather as much straightforward as possible

Reply:
Removed this method, resolving